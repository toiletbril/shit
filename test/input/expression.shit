-7+(2^2+3*2|(5|2)-1>2)+-(-((46+2)*3^2)-(2<<25/5))+5^2--(52*(25%10))-(2>=2||5>>24)+(-(+(~(2)+((1<=2))>>((1<2)/4%2)))+1!=0)-(5+10*2)-6-~5
