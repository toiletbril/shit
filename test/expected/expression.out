[Binary ^]
  [Binary +]
    [Binary +]
      [Binary +]
        [Unary -]
          [Number 7]
        [Binary |]
          [Binary ^]
            [Number 2]
            [Binary +]
              [Number 2]
              [Binary *]
                [Number 3]
                [Number 2]
          [Binary >]
            [Binary -]
              [Binary |]
                [Number 5]
                [Number 2]
              [Number 1]
            [Number 2]
      [Unary -]
        [Binary -]
          [Unary -]
            [Binary ^]
              [Binary *]
                [Binary +]
                  [Number 46]
                  [Number 2]
                [Number 3]
              [Number 2]
          [Binary <<]
            [Number 2]
            [Binary /]
              [Number 25]
              [Number 5]
    [Number 5]
  [Binary -]
    [Binary -]
      [Binary -]
        [Binary +]
          [Binary -]
            [Binary -]
              [Number 2]
              [Unary -]
                [Binary *]
                  [Number 52]
                  [Binary %]
                    [Number 25]
                    [Number 10]
            [Binary ||]
              [Binary >=]
                [Number 2]
                [Number 2]
              [Binary >>]
                [Number 5]
                [Number 24]
          [Binary !=]
            [Binary +]
              [Unary -]
                [Unary +]
                  [Binary >>]
                    [Binary +]
                      [Unary ~]
                        [Number 2]
                      [Binary <=]
                        [Number 1]
                        [Number 2]
                    [Binary %]
                      [Binary /]
                        [Binary <]
                          [Number 1]
                          [Number 2]
                        [Number 4]
                      [Number 2]
              [Number 1]
            [Number 0]
        [Binary +]
          [Number 5]
          [Binary *]
            [Number 10]
            [Number 2]
      [Number 6]
    [Unary ~]
      [Number 5]
54
